reporting_endpoint.py

```diff
@@ -109,107 +76,63 @@ def report_user_balance():
-    if frequency == 'month':
-        return redirect(str(paydbtransactions_url)+str(paydbtransactions_filter)+str(FIRST_DAY_CURRENT_MONTH)+'+to+'+str(FIRST_DAY_NEXT_MONTH))
-    if frequency == 'year':
-        return redirect(str(paydbtransactions_url)+str(paydbtransactions_filter)+str(FIRST_DAY_CURRENT_YEAR)+'+to+'+str(FIRST_DAY_NEXT_YEAR))
-    return redirect(str(paydbtransactions_url))
-
-def report_proposal_txs(frequency):
-    proposal_url = str('/admin/rewards')
-    proposal_filter = str('?flt0_0=')
-    if frequency == 'lifetime':
-        return redirect(str(proposal_url))
-    if frequency == 'today':
-        return redirect(str(proposal_url)+str(proposal_filter)+str(TODAY)+'+to+'+str(TOMORROW))
-    if frequency == 'yesterday':
-        return redirect(str(proposal_url)+str(proposal_filter)+str(YESTERDAY)+'+to+'+str(TODAY))
-    if frequency == 'week':
-        return redirect(str(proposal_url)+str(proposal_filter)+str(MONDAY)+'+to+'+str(NEXT_MONDAY))
-    if frequency == 'month':
-        return redirect(str(proposal_url)+str(proposal_filter)+str(FIRST_DAY_CURRENT_MONTH)+'+to+'+str(FIRST_DAY_NEXT_MONTH))
-    if frequency == 'year':
-        return redirect(str(proposal_url)+str(proposal_filter)+str(FIRST_DAY_CURRENT_YEAR)+'+to+'+str(FIRST_DAY_NEXT_YEAR))
-    return redirect(str(proposal_url))
-
-def claimed_proposal_payment(table1, table2, start_date, end_date):
-    result = table1.query.join(table2, table1.id==table2.reward_proposal_id)\
-            .filter(and_(table1.date_authorized >= str(start_date),\
-            table1.date_authorized < str(end_date))).filter(table2.status == 'sent_funds').with_entities(func.sum(table2.amount)).scalar()
+def user_counting(table, start_date, end_date):
+    result = table.query.filter(and_(table.confirmed_at >= str(start_date), table.confirmed_at <= str(end_date))).count()
     if not result:
         result = 0
-    result = utils.int2asset(result)
     return result
 
-def unclaimed_proposal_payment(table1, table2, start_date, end_date):
-    result = table1.query.join(table2, table1.id==table2.reward_proposal_id)\
-            .filter(and_(table1.date_authorized >= str(start_date),\
-            table1.date_authorized < str(end_date))).filter(table2.status != 'sent_funds').with_entities(func.sum(table2.amount)).scalar()
+def transaction_count(table, start_date, end_date):
+    result = table.query.filter(and_(table.date >= str(start_date), table.date < str(end_date))).count()
     if not result:
         result = 0
-    result = utils.int2asset(result)
     return result
 
-def authorized_unclaimed_payment_proposal(table1, table2):
-    result = table1.query.join(table2, table1.id==table2.reward_proposal_id)\
-            .filter(table2.status != 'sent_funds').filter(table1.status == 'authorized')\
-            .with_entities(func.sum(table2.amount)).scalar()
-    if not result:
-        result = 0
-    return result
+def get_decimals(asset):
```

Comment:
you can get rid of this function and just use `dasset.asset_int_to_dec` instead

Reply:
used the above